index.js

```diff
@@ -20,14 +23,22 @@ class ImmediateStore {
 
   get (index, opts, cb) {
     if (typeof opts === 'function') return this.get(index, null, opts)
+    if (!opts) opts = {}
+
+    const memoryBuffer = this.mem[index]
+
+    // if the chunk isn't in the immediate memory cache
+    if (!memoryBuffer) {
+      return this.store.get(index, opts, cb)
+    }
 
-    const start = (opts && opts.offset) || 0
-    const end = opts && opts.length && (start + opts.length)
+    const start = opts.offset || 0
+    const end = opts.length ? (start + opts.length) : memoryBuffer.length
 
-    const buf = this.mem[index]
-    if (buf) return nextTick(cb, null, opts ? buf.slice(start, end) : buf)
+    const slicedBuffer = memoryBuffer.slice(start, end)
```

Comment:
This always slices the buffer, which does a copy, even when there's no `start` or `end`.